docs/performance/inserts/tuning.txt

```diff
@@ -0,0 +1,172 @@
+.. _config_tuning:
+
+================================
+Configuration Tuning for Inserts
+================================
+
+This document outlines a number of hardware and software configuration changes
+you can make to tune your setup for inserts performance.
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+Hardware
+========
+
+Solid-State Drives
+------------------
+
+Switch to *Solid-State Drives* (SSDs) if you can.
+
+SSDs are generally much faster than hard-disk drives, and usually offer the
+most cost-effective hardware upgrade for a CrateDB setup.
+
+Use the ``iostat`` tool to display disk I/O statistics, which should include
+output like this:
+
+.. code-block:: text
+
+    sh$ iostat
+    ...
+    avg-cpu:  %user   %nice %system %iowait  %steal   %idle
+               0.53    0.01    0.05    0.01    0.00   99.41
+    ...
+
+
+Here, the most interesting metric is ``%iowait``. This tells you how
+time the CPU is waiting for disk I/O. You can expect better insert performance
+from CrateDB relative to how low this value is.
+
+CPUs
+----
+
+If your queries are complex, the additional CPU overhead during the
+:ref:`analyzing phase <parsing_analyzing>` of insert query processing might
+result in a CPU performance bottleneck.
+
+If this becomes the case, and simplifying your queries isn't an option,
+upgrading your CPUs will be a cost-effective way to boost CrateDB performance.
+
+RAM Disks
+---------
+
+TODO are there any circumstances under which we would advise using a ram disk
+
+Software
+========
+
+Server
+------
+
+Replicas
+........
+
+Biggest influence: replicas (esp 0 vs . 1+)
+
+(Because primary write first and only after it writes the replicas)
+
+TODO: need details. what are we recommending here?
```

Comment:
0 should almost never be used (except maybe temporary for initial imports) 
At least 1 for a production 3 cluster setup.

More depends on the expected availability. (And going from 1 to 2 isn't as costly as from 0 to 1 in terms of insert duration, load increases linearly)